elide-core/src/main/java/com/yahoo/elide/security/User.java

```diff
@@ -20,13 +19,11 @@
  */
 public class User {
     @Getter private final Object opaqueUser;
-    private final Map<Class<? extends Check>, UserPermission> okUserPermissions;
-    private final Map<Class<? extends Check>, Map<PersistentResource, Boolean>> okCheckResources;
```

Comment:
We need to discuss this more offline. If we allow side-effects in checks, caching may throw off those semantics. However, side-effects are generally a horrible thing (in my humble opinion :)) so I'm happy to reinstate the cache after some more thought.